boost_histogram/_internal/hist.py

```diff
@@ -27,10 +27,11 @@ def _arg_shortcut(item):
     elif isinstance(item, _Axis):
         return item._ax
     else:
-        return item
         # TODO: This currently support raw axis object for old tests.
         # Replace with:
         # raise TypeError("Only axes supported in histogram constructor")
+        # TODO: Currently segfaults if we pass in a non-axis to the C++ code
```

Comment:
Why is this segfaulting?

Reply:
I accidentally wrote `super(histogram, self).__init__(self, *args, **kargs)`, which caused a Segfault. After (quite a bit of) digging around, I found that PyBind11 was trying to prepare an an error message, since the wrapper "`histogram`" was not a valid argument to `histogram`, but in doing so, it was calling `__repr__` on `histogram`, and since the repr of histogram expected `self._hist` to be available, it was throwing a Python error, which PyBind11 was not catching, and the uncaught error caused a segfault.

I think it's something that can be fixed in PyBind11; making an error message shouldn't segfault if the item in the error message has a faulty repr.

Reply:
(The non-axis must also have a broken `__repr__`)